pootle/apps/pootle_store/unit/timeline.py

```diff
@@ -45,173 +45,204 @@ class UnitTimelineLog(UnitLog):
+                entry['translation'] = True
 
-    def suggestion_entry(self):
-        entry = self.entry_dict
-        entry.update({'suggestion_text': self.submission.suggestion_target})
         return entry
 
+
+class UnitStateChangedEntry(SubmissionEntry):
+
     @property
     def entry(self):
-        if self.submission.field == SubmissionFields.STATE:
-            return self.state_change_entry()
-        elif self.submission.suggestion:
-            return self.suggestion_entry()
-        elif self.submission.qc_name:
-            return self.qc_entry()
-        return self.base_entry()
+        params = {
+            'old_value': STATES_MAP[int(to_python(self.submission.old_value))],
+            'new_value': STATES_MAP[int(to_python(self.submission.new_value))]}
+        value = format_html(_(
+            u"%(old_value)s "
+            u"<span class='timeline-arrow'></span> "
+            u"%(new_value)s", params))
 
+        return dict(value=value,
+                    state=True)
 
-class Timeline(object):
 
-    entry_class = TimelineEntry
-    fields = SubmissionProxy.timeline_fields
+class CommentUpdatedEntry(SubmissionEntry):
+    @property
+    def entry(self):
+        if self.submission.new_value:
+            return dict(
+                value=self.submission.new_value,
+                sidetitle=_("Comment:"),
+                comment=True)
+
+        return dict(description=_("Last comment removed"))
+
 
-    def __init__(self, ob):
-        self.object = ob
+class CheckEntry(SubmissionEntry):
+    @cached_property
+    def check_name(self):
+        return self.submission.quality_check.name
+
+    @cached_property
+    def check_url(self):
+        return u''.join(
+            [reverse('pootle-checks-descriptions'),
+             '#', self.check_name])
 
     @property
-    def grouped_entries(self):
-        grouped_entries = self.get_grouped_entries()
-        grouped_entries = self.add_creation_entry(grouped_entries)
-        grouped_entries.reverse()
-        return grouped_entries
+    def check_link(self):
+        return format_html("<a href='{}'>{}</a>", self.check_url,
+                           CHECK_NAMES[self.check_name])
+
 
+class CheckMutedEntry(CheckEntry):
     @property
-    def submissions(self):
-        submission_filter = (
-            Q(field__in=[SubmissionFields.TARGET, SubmissionFields.STATE,
-                         SubmissionFields.COMMENT, SubmissionFields.NONE]))
-        subs = (
-            Submission.objects.filter(unit=self.object)
-                              .filter(submission_filter))
-        if self.object.changed and self.object.change.commented_on:
-            subs = subs.exclude(
-                field=SubmissionFields.COMMENT,
-                creation_time=self.object.change.commented_on)
-        return subs.order_by("id")
+    def entry(self):
+        return dict(
+            description=format_html(_(
+                "Muted %(check_link)s check",
```

Comment:
I think this will confuse translators too, so perhaps better use "Muted %(check_name)s check".